### ISSUE TITLE
The current git17.RemoteContains doesn't appear to work with git 2.7.4

### ISSUE THREAD

**Issue by siebenmann** [shurcooL/vcsstate#12]
Ubuntu 16.04 has git 2.7.4, which is before git 2.8 and so causes vcsstate to use git17 to handle things. git17.RemoteContains winds up running `git branch -r --contains <rev> origin/master`. In git 2.7.4, this reports:

````
$ git branch -r --contains ff45d5492410f9f4be8e749dba8ecfebf635ed77 origin/master
  origin/master
````

git17.RemoteContains expects it to be `origin/HEAD` instead. I suspect that this is fallout from the change in commit ff45d549, 'Add default branch parameter to RemoteContains', and that possibly RemoteContains should now be checking against origin/<defaultBranch> instead of origin/HEAD.

**Comment by dmitshur** [comment]
Hi @siebenmann, thanks for reporting this.

Can you elaborate on the context? Are you using `vcsstate` as part of your own application, or are you having issues with `gostatus` or `Go-Package-Store` which use this package?

The change in ff45d54 made the remote default branch a parameter rather than hardcoding it to be `HEAD` (see commit message for rationale). Of course, it's still possible to pass `"HEAD"` as the last argument, which would produce behavior identical to before that commit.

**Comment by siebenmann** [comment]
I'm having issues with both `gostatus` and `Go-Package-Store` on Ubuntu 16.04 as a result of this issue.  I get stuff that looks like this:

````
$ Go-Package-Store
[...]
skipping "github.com/shurcooL/octiconssvg" because:
        local revision "1aed2117d2aa8eed76a052936e468e5354750701" is ahead of remote revision "8c9861b86a08c72d14e0285d0dc313bb6df52295"
[...]
$ gostatus github.com/shurcooL/octiconssvg
  ±  github.com/shurcooL/octiconssvg/...
        ± Update available; local revision is ahead of remote revision
````

I don't rebuild or run either command frequently on our Ubuntu 16.04 machines so I can't be absolutely sure that this is due to ff45d549, but I'm pretty sure I've rebuilt `Go-Package-Store` and `gostatus` within the past month or two and they worked then.

**Comment by dmitshur** [comment]
I see, thanks. Can you do me a favor and try the following first:

```
go get -u github.com/shurcooL/Go-Package-Store/cmd/Go-Package-Store
go get -u github.com/shurcooL/gostatus
```

Then I can be sure you're testing with the very latest versions.

If that doesn't resolve the issue, I'll need to investigate further.

**Comment by siebenmann** [comment]
I've done these `go get -u` updates and verified that this happens with the latest versions (with the same errors).

(I should have been clearer in my description; I had already done these as part of my updates earlier today, when I noticed the issue, because my usual update process is to use `go get -u` on a collection of commonly-used commands and then run `Go-Package-Store` to look for anything else. What I wasn't certain of is how recently before today I had gone through this process, so I can't say definitely that this stopped working very recently.)

**Comment by dmitshur** [comment]
All right, thanks. To resolve this, I'll need to find a way to reproduce it.

I think I have enough information for now to be able to do that, so I'll report back with what I find.

**Comment by dmitshur** [comment]
I am able to reproduce the issue with `git17` backend (quite easily). Will work on a fix. :)

**Comment by dmitshur** [comment]
I suspect I see where the bug is (and it's quite obvious now). I forgot to update these lines for `git17` backend when I made the change in ff45d5492410f9f4be8e749dba8ecfebf635ed77:

https://github.com/shurcooL/vcsstate/blob/ff45d5492410f9f4be8e749dba8ecfebf635ed77/git17.go#L111-L112

I didn't forget for `git28` backend because I actually use that one myself.

**Comment by dmitshur** [comment]
This should be fixed in fe5057596735a6385fde3b223ad822be3455f3d7.

Thanks a lot for this invaluable issue report! Please don't hesitate to let me know if run into any other issues.

**Comment by siebenmann** [comment]
I just rebuilt gostatus and Go-Package-Store with `go get -u` and retested both of them, so I can confirm that this is fixed now. Thank you for the fast fix!
